gpjax/decision_making/acquisition_maximizer.py

```diff
@@ -0,0 +1,127 @@
+        to evaluate at `query_points`.
+
+    Returns:
+        Float[Array, "1 D"]: Point in `query_points` which maximises the acquisition
+        function.
+    """
+    acquisition_function_values = acquisition_function(query_points)
+    max_acquisition_function_value_idx = jnp.argmax(
+        acquisition_function_values, axis=0, keepdims=True
+    )
+    best_sample_point = jnp.take_along_axis(
+        query_points, max_acquisition_function_value_idx, axis=0
+    )
+    return best_sample_point
+
+
+@dataclass
+class AbstractAcquisitionMaximizer(ABC):
+    """Abstract base class for acquisition function maximizers."""
+
+    @abstractmethod
+    def maximize(
+        self,
+        acquisition_function: AcquisitionFunction,
+        search_space: AbstractSearchSpace,
+        key: KeyArray,
+    ) -> Float[Array, "1 D"]:
+        """Maximize the given acquisition function over the search space provided.
+
+        Args:
+            acquisition_function (AcquisitionFunction): Acquisition function to be
+            maximized.
+            search_space (AbstractSearchSpace): Search space over which to maximize
+            the acquisition function.
+            key (KeyArray): JAX PRNG key.
+
+        Returns:
+            Float[Array, "1 D"]: Point at which the acquisition function is maximized.
+        """
+        raise NotImplementedError
+
+
+@dataclass
+class ContinuousAcquisitionMaximizer(AbstractAcquisitionMaximizer):
+    """The `ContinuousAcquisitionMaximizer` class is used to maximize acquisition
+    functions over the continuous domain with L-BFGS-B. First we sample the acquisition
+    function at `num_initial_samples` points from the search space, and then we run
+    L-BFGS-B from the best of these initial points.
+    """
+
+    num_initial_samples: int
+
+    def __post_init__(self):
+        if self.num_initial_samples < 1:
+            raise ValueError(
+                f"num_initial_samples must be greater than 0, got {self.num_initial_samples}."
+            )
+
+    def maximize(
+        self,
+        acquisition_function: AcquisitionFunction,
+        search_space: ContinuousSearchSpace,
+        key: KeyArray,
+    ) -> Float[Array, "1 D"]:
+        initial_sample_points = search_space.sample(self.num_initial_samples, key=key)
+        best_initial_sample_point = _get_discrete_maximizer(
+            initial_sample_points, acquisition_function
+        )
+
+        # Jaxopt minimizer requires a function which returns a scalar. It calls the
+        # acquisition function with one point at a time, so the acquisition function
+        # returns an array of shape [1, 1], so  we index to return a scalar. Note that
+        # we also return the negative of the acquisition function - this is because
+        # acquisition functions should be *maximimized* but the Jaxopt minimizer
+        # minimizes functions.
+        def scalar_acquisition_fn(x: Float[Array, "1 D"]) -> ScalarFloat:
```

Comment:
Maybe add an underscore before this function to make it explicit that it's private to the scope of the method